modules/xmpp/ChatRoom.js

```diff
@@ -510,10 +513,14 @@ ChatRoom.prototype.onPresenceUnavailable = function (pres, from) {
             delete this.members[i];
             this.onParticipantLeft(i, member.isFocus);
         }
+        this.connection.emuc.doLeave(this.roomjid);
+        this.eventEmitter.emit(XMPPEvents.MUC_LEFT);
```

Comment:
Changed that, so now we process the 110 as we did before and fire muc_left only if it is not kick and then process the kick part.